lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -1378,27 +1379,34 @@ protected function getSelectManyToManyJoinSQL(array $manyToMany)
         return ' INNER JOIN ' . $joinTableName . ' ON ' . implode(' AND ', $conditions);
     }
 
+    public function getMaxBatchedInsertQueries()
+    {
+        $idGenerator    = $this->class->idGenerator;
+        $isPostInsertId = $idGenerator->isPostInsertGenerator();
+
+        if ($isPostInsertId || $this->class->isVersioned) {
+            return 1;
```

Comment:
We can start naively here with 1, but we could be smarter depending of post generated identifier. If using auto incremented, certain RDBMS (such as MySQL and MariaDB, very likely PgSQL too) guarantee the ordering of incremental identifiers on a per transaction basis. This means in case of concurrent writes, the first transaction would be 1 - X as ID incrementally, and second transaction would generate X+1 - Y, also incrementally.

But as an initial implementation, this is more than enough.